### ISSUE TITLE
Unable to compile flutter app with Smartech_flutter_plugin 2.0.0 on flutter 3.0.0

### ISSUE THREAD

**Issue by dipesh-panchal** [NetcoreSolutions/Smartech-Flutter-Plugin#4]
My app was running perfectly for flutter 2.10.5 with Smartech_flutter_plugin 2.0.0, but since my upgrade to flutter 3.0.0 I'm facing this error: `SmartechPlugin.kt: (241, 17): 'error' overrides nothing`. 
I tried to roll back to flutter 2.10.5 and it is working fine.
@manish-netcore @biscottis 
below is the extended log  :

```
/Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/store_redirect-2.0.1/android/src/main/java/com/danieldallos/storeredirect/StoreRedirectPlugin.java:29: warning: [deprecation] Registrar in PluginRegistry has been deprecated
  public static void registerWith(Registrar registrar) {
                                  ^
2 warnings
e: /Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/smartech_flutter_plugin-2.0.0/android/src/main/kotlin/com/smartech/flutter/smartech_plugin/SmartechPlugin.kt: (58, 31): Type mismatch: inferred type is BinaryMessenger? but BinaryMessenger was expected
e: /Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/smartech_flutter_plugin-2.0.0/android/src/main/kotlin/com/smartech/flutter/smartech_plugin/SmartechPlugin.kt: (235, 67): Object is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result
e: /Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/smartech_flutter_plugin-2.0.0/android/src/main/kotlin/com/smartech/flutter/smartech_plugin/`SmartechPlugin.kt: (241, 17): 'error' overrides nothing`
e: /Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/smartech_flutter_plugin-2.0.0/android/src/main/kotlin/com/smartech/flutter/smartech_plugin/SmartechPlugin.kt: (273, 65): Object is not abstract and does not implement abstract member public abstract fun error(p0: String, p1: String?, p2: Any?): Unit defined in io.flutter.plugin.common.MethodChannel.Result
e: /Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/smartech_flutter_plugin-2.0.0/android/src/main/kotlin/com/smartech/flutter/smartech_plugin/SmartechPlugin.kt: (279, 15): 'error' overrides nothing
/Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/user_experior-2.1.6/android/src/main/java/com/userexperior/user_experior/UserExperiorPlugin.java:16: warning: [deprecation] Registrar in PluginRegistry has been deprecated
import io.flutter.plugin.common.PluginRegistry.Registrar;
                                              ^
/Users/xxxxxx/FlutterSDK/flutterLatest/.pub-cache/hosted/pub.dartlang.org/user_experior-2.1
.
.
.
.
.
3 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':smartech_flutter_plugin:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 26s
Exception: Gradle task assembleTejimandiBetaDebug failed with exit code 1


```

**Comment by icodebuster** [comment]
We are working on the same. For time being use Flutter SDK `v2.10.5`

**Comment by icodebuster** [comment]
Released in version [v2.0.1](https://github.com/NetcoreSolutions/Smartech-Flutter-Plugin/releases/tag/v2.0.1)
